great_expectations/data_context/data_context.py

```diff
@@ -3654,11 +3654,11 @@ def _test_instantiation_of_profiler_from_yaml_config(
         """
         print(f"\tInstantiating as a {class_name}, since class_name is {class_name}")
 
-        profiler_name = name or config.get("name") or "my_temp_profiler"
+        profiler_name: str = name or config.get("name") or "my_temp_profiler"
 
-        profiler_config = RuleBasedProfilerConfig.from_commented_map(
-            commented_map=config
-        )
+        profiler_config: Union[
+            RuleBasedProfilerConfig, dict
+        ] = RuleBasedProfilerConfig.from_commented_map(commented_map=config)
```

Comment:
Why a union type here? Is this to provide context to the result dict object?

Reply:
@cdkini We do not have to do it this way -- we could use two different variables, but for now, it is is because `from_commented_map()` returns the typed object, but then we set the same variable to the output of `to_json_dict()` -- I am OK with leaving as is.

Reply:
I'd err on the side of immutable types but I'm okay with this for the time being. Thanks!